app/license.go

```diff
@@ -124,14 +125,24 @@ func (s *Server) SaveLicense(licenseBytes []byte) (*model.License, *model.AppErr
 	s.ReloadConfig()
 	s.InvalidateAllCaches()
 
-	// start job server if necessary - this handles the edge case where a license file is uploaded, but the job server
+	// restart job server workers - this handles the edge case where a license file is uploaded, but the job server
 	// doesn't start until the server is restarted, which prevents the 'run job now' buttons in system console from
 	// functioning as expected
-	if *s.Config().JobSettings.RunJobs && s.Jobs != nil && s.Jobs.Workers != nil {
-		s.Jobs.StartWorkers()
+	if *s.Config().JobSettings.RunJobs && s.Jobs != nil {
+		if err := s.Jobs.StopWorkers(); err != nil && !errors.Is(err, jobs.ErrWorkersNotRunning) {
+			mlog.Warn("Stopping job server workers failed", mlog.Err(err))
+		}
+		if err := s.Jobs.InitWorkers(); err != nil {
+			mlog.Warn("Initializing job server workers failed", mlog.Err(err))
+		}
+		if err := s.Jobs.StartWorkers(); err != nil {
+			mlog.Warn("Starting job server workers failed", mlog.Err(err))
+		}
```

Comment:
This is a slight change in application logic. Before we were essentially re-initializing (and leaking) and starting new workers but nothing would get stopped. Here to avoid the leak we first stop the workers, re-initialize and start them again. 
Depending on the expectations this might be okay. If not, we would need to add a more complex state to keep track of every single worker and only start those that are not yet running.

Reply:
Is this to handle the case of multiple license uploads? Is there a need to restart a job server again? Can we not keep it running if it's already running?

Reply:
There's a need cause of enterprise only jobs that could be not running before adding the license.